src/python_inspector/resolution.py

```diff
@@ -453,16 +454,22 @@ def get_candidates(
         """
         Generate candidates for the given identifier. Overridden.
         """
+        valid_versions = []
         for version in all_versions:
             parsed_version = parse_version(version)
-            if not is_valid_version(
+            if is_valid_version(
                 parsed_version=parsed_version,
                 requirements=requirements,
                 identifier=identifier,
                 bad_versions=bad_versions,
             ):
-                continue
-            yield Candidate(name=name, version=parsed_version, extras=extras)
+                valid_versions.append(parsed_version)
+        if all(version.is_prerelease for version in valid_versions):
```

Comment:
What about removing the pass below and using this instead?

```suggestion
        if not all(version.is_prerelease for version in valid_versions):
```

or may be there is a way to better express your intentions?